.editorconfig

```diff
@@ -1,9 +1,26 @@
+# EditorConfig file: http://editorconfig.org/
+
 root = true
 
-[*.{c,h,d,di,dd}]
+# Default settings for all file types:
+# + save files in UTF-8 encoding
+# + use Unix line-endings and a final newline
+# + remove whitespace at the end of lines
+[*]
+charset = utf-8
 end_of_line = lf
 insert_final_newline = true
+trim_trailing_whitespace = true
+
+# File type specific settings:
+[*.{asm,S,c,cpp,m,h,d,di,dd,ddoc,dt,json,sdl,R,bat,bash,fish,sh,zsh}]
```

Comment:
I basically ran `find my-dlang-repos-folder/ -type f -exec sh -c 'echo "${0##*.}"' {} \; | sort | uniq > file_types` to get a list of all file types and then added here what I thought was sensible.